### PR TITLE
Tensosensor calibration with steps

### PR DESCRIPTION
**Pull request by Minyewoo**

### CODE REVIEW

File: lib/pages/failures/widgets/failures_body.dart [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    }
+    _columnsVisibility = Map.fromEntries(
+      _columns.keys.map((signal) => MapEntry(signal, true)),
+    );
+    super.initState();
+  }
+  //
+  @override
+  Widget build(BuildContext context) {
+    const timeColumnSize = 230.0;
```
Comment (a-givertzman) [inline lib/pages/failures/widgets/failures_body.dart:68]: please remove hard-coded literals

File: lib/pages/tensosensor_calibration/widgets/tensosensor_calibration_step.dart [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    _viewIndex = viewIndex,
+    _fieldData = fieldData,
+    _stepType = stepType,
+    _onCancel = onCancel, 
+    _onFinish = onFinish, 
+    _onNext = onNext;
+  //
+  @override
+  Widget build(BuildContext context) {
+    const itemWidth = 150.0;
```
Comment (a-givertzman) [inline lib/pages/tensosensor_calibration/widgets/tensosensor_calibration_step.dart:120]: can we pass it thru constructor params?

File: lib/pages/work_cycles/widgets/work_cycles_app_bar.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  }) : 
+    _endingTime = endingTime, 
+    _beginningTime = beginningTime;
+  //
+  @override
+  Widget build(BuildContext context) {
+    const dateFieldWidth = 220.0;
+    return AppBarWidget(
+      title: const Localized('Work cycles').v,
+      height: kTextTabBarHeight * 1.5,
```
Comment (a-givertzman) [inline lib/pages/work_cycles/widgets/work_cycles_app_bar.dart:23]: again it strange var or const: kTextTabBarHeight, can we avoid using it?

File: lib/pages/work_cycles/widgets/work_cycles_app_bar.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    super.key,
+    DateTime? beginningTime,
+    DateTime? endingTime, 
+  }) : 
+    _endingTime = endingTime, 
+    _beginningTime = beginningTime;
+  //
+  @override
+  Widget build(BuildContext context) {
+    const dateFieldWidth = 220.0;
```
Comment (a-givertzman) [inline lib/pages/work_cycles/widgets/work_cycles_app_bar.dart:20]: bay be more suitable to pass it thru constructor?

**Review by a-givertzman — APPROVED** [review]
try to fix commented points where it can be reached direct & fast, then you can merge current brunch

File: lib/core/widgets/field/submitable_field.dart
[the remark this answers is not in the document]
Comment (Minyewoo) [reply lib/core/widgets/field/submitable_field.dart:1]: It's used in WorkCyclesAppBar and we may use it in TensosensorCalibrationStep in the future.

**Pull request merged**